### PR TITLE
Add Nacos back into SkyWalking as an option(nacos-1.3.0)

### PR DESCRIPTION
**Pull request by songzhendong**
Add Nacos back into SkyWalking as an option(nacos-1.3.0)

### CODE REVIEW

File: dist-material/release-docs/NOTICE [hunk 1]
```diff
@@ -838,7 +838,43 @@ from and not be held liable to the user for any such damages as noted
 above as far as the program is concerned.
 
 ------
+===========================================================================
+nacos-1.3.0 Notice
+===========================================================================
+Nacos
+Copyright 2018-2019 The Apache Software Foundation
```
Comment (wu-sheng) [inline dist-material/release-docs/NOTICE:845]: Nacos is not belong to the ASF, the NOTICE is wrong.

File: dist-material/release-docs/NOTICE [hunk 2]
```diff
@@ -838,7 +838,43 @@ from and not be held liable to the user for any such damages as noted
...
+
+This product includes software developed at
+The Alibaba MiddleWare Group.
+
+------
+This product has a bundle Spring Boot:
```
Comment (wu-sheng) [inline dist-material/release-docs/NOTICE:851]: We don't bundle with Spring Boot, maybe Nacos server is. But Notice base on what we used.

So basically, you could remove the NOTICE change.

File: docs/en/setup/backend/backend-cluster.md [hunk 3]
```diff
@@ -82,6 +83,14 @@ The following setting are provided to set the host and port manually, based on y
 - internalComHost, the host registered and other oap node use this to communicate with current node.
 - internalComPort, the port registered and other oap node use this to communicate with current node.
 
+## Nacos
+Set the **cluster/selector** to **nacos** in the yml to enable.
+
+```yaml
+cluster:
+  selector: ${SW_CLUSTER:nacos}
+  # other configurations
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-cluster.md:92]: Please list the configuration of Nacos, and explain them. Also, Nacos should be at the last one.

File: docs/en/setup/backend/backend-cluster.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ with each other.
 - [Kubernetes](#kubernetes). When backend cluster are deployed inside kubernetes, you could choose this
 by using k8s native APIs to manage cluster.
 - [Consul](#consul). Use Consul as backend cluster management implementor, to coordinate backend instances.
+- [Nacos](#nacos). Use Nacos to coordinate backend instances.
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-cluster.md:12]: Move to the last one.

**Comment by KomachiSion** [comment]
NOTICE of Nacos has changed.

File: dist-material/release-docs/NOTICE [hunk 5]
```diff
@@ -838,7 +838,43 @@ from and not be held liable to the user for any such damages as noted
...
 
 ------
+===========================================================================
+nacos-1.3.0 Notice
+===========================================================================
+Nacos
+Copyright 2018-2019 The Apache Software Foundation
+
+This product includes software developed at
+The Alibaba MiddleWare Group.
```
Comment (wu-sheng) [inline dist-material/release-docs/NOTICE:848]: @songzhendong If you want to keep the NOTICE(we could keep). We just need the above part. 
Please take a look the latest version, I have sent a PR to fix it. 

The things below we don't use, so, don't need to add.

**Comment by wu-sheng** [comment]
Seem Nacos Server doesn't start up successfully in the IT, please recheck what is happening. The test could pass in the Mac and Windows platforms.

**Comment by wu-sheng** [comment]
@KomachiSion Do you have any recommendation/configruation for Nacos Docker image running in the integration test based on Maven?

**Comment by KomachiSion** [comment]
I see the CI log, Nocos has started successfully with standalone mode and embedded storage. I think it's the best configuration for IT.

And I see the error is timeout, I'm not sure the reason whether is port mapping or program.

**Comment by wu-sheng** [comment]
The timeout seems notification timeout, the UT can't get the notification from Nacos server. @KangZhiDong

**Comment by KomachiSion** [comment]
I will clone the source repo and have a check.

**Comment by wu-sheng** [comment]
@KomachiSion We could pass the test locally on MacOS from time to time, but on the GitHub action, maybe with limited source, it fails in high frequency, it is very strange.

**Comment by kezhenxu94** [comment]
> @KomachiSion We could pass the test locally on MacOS from time to time, but on the GitHub action, maybe with limited source, it fails in high frequency, it is very strange.

Does Nacos fail in high frequency? I didn’t notice this before

**Comment by KomachiSion** [comment]
When I debug, I find that nacos has notify to skywalking `NacosConfigWatcherRegister` but the `testValue` in `NacosConfigurationTestProvider` do not changed.

**Comment by KomachiSion** [comment]
image

**Comment by KomachiSion** [comment]
I find out the reason, the delete config has no notify event from nacos. The problem has fixed, but need waiting for 1.3.1 release.

We has plan to release 1.3.1 next week, and this week will release 1.3.1-BETA for community.

So I think we can pending this PR for one or two weeks.

Sorry for troubles.

**Comment by wu-sheng** [comment]
@KomachiSion We could wait for your official 1.3.1 release. Next SkyWalking release should be near the end of next month.

**Comment by wu-sheng** [comment]
@songzhendong Please be advised. We need to wait for another week :)

It is great we could have help from the original team.

**Comment by KomachiSion** [comment]
@wu-sheng @songzhendong Nacos 1.3.1 has released.

**Comment by wu-sheng** [comment]
@KomachiSion Noticed, have asked @songzhendong  to do some tests and update this PR.

File: dist-material/release-docs/LICENSE [hunk 6]
```diff
@@ -311,6 +311,7 @@ The text of each license is the standard Apache 2.0 license.
     consul-client 1.2.6: https://github.com/rickfast/consul-client, Apache 2.0
     okhttp 3.9.0: https://github.com/square/okhttp, Apache 2.0
     prometheus client_java 0.6.0: https://github.com/prometheus/client_java, Apache 2.0
+    nacos 1.3.0: https://github.com/alibaba/nacos, Apache 2.0
```
Comment (kezhenxu94) [inline dist-material/release-docs/LICENSE:314]:
```suggestion
    nacos 1.3.1: https://github.com/alibaba/nacos, Apache 2.0
```

**Comment by wu-sheng** [comment]
This PR has issues about merging upstream. Should be fixed later.

**Comment by songzhendong** [comment]
I'll check these mistakes

File: dist-material/release-docs/NOTICE (hunk 1 above)
Comment (songzhendong) [reply dist-material/release-docs/NOTICE]: > The PR becomes messy and removes lots of commits from the upstream, I suggest to rebase or simply revert #4867 first and then upgrade the Nacos version immediately, the latter would be more friendly to reviewers

thanks for reviews

File: oap-server/pom.xml [hunk 7]
```diff
@@ -389,6 +389,16 @@
...
+            <dependency>
+                <groupId>com.alibaba.nacos</groupId>
+                <artifactId>nacos-common</artifactId>
+                <version>${nacos.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.alibaba.nacos</groupId>
+                <artifactId>nacos-api</artifactId>
+                <version>${nacos.version}</version>
+            </dependency>
```
[the remark this answers is not in the document]
Comment (songzhendong) [reply oap-server/pom.xml]: > @wu-sheng @songzhendong Nacos 1.3.1 has released.

thanks

**Comment by wu-sheng** [comment]
Replaced by #5137

**Pull request closed without merging**